src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/equalsavoidnull/InputEqualsAvoidNull.java

```diff
@@ -427,3 +427,12 @@ int getInt() {
     }
 }
 
+class TestThisWithNotStringInstance {
+
+    MyString notString;
+
+    void foo() {
+        this.notString.equals(""); // ok
```

Comment:
I might not something, but it should be violation https://checkstyle.org/config_coding.html#EqualsAvoidNull . please explain.

Reply:
@romani If it should be a violation it is not an issue from this PR as I can reproduce it with released versions.
````
$ cat TestClass.java
public class TestClass {

    MyString notString;

    void method() {
        this.notString.equals(""); // ok
    }

}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="EqualsAvoidNull"/>
    </module>
</module>

$ java -jar checkstyle-8.24-all.jar -c TestConfig.xml TestClass.java
Starting audit...
Audit done.
````

I got the same result all the way back to version 6.0 .

Reply:
I tried removing `this.` and version 6.9 got a violation but 6.10 and after didn't.

Reply:
@romani I just happened to notice the type of the variable.
`notString` is `MyString` and not `String`. This is why there is no violation. If you change the type to `String` a violation comes up.

Reply:
Yes, there is a call to `MyString::equals(String s)` which **can** return `true`.
After replacement, there will be a call to `String:: equals(Object o)`, which **will definitely** return `false` for any non-string objects.

Reply:
Yes, my bad, forgot that Check is String specific